share/storage.go

```diff
@@ -102,7 +102,9 @@ func (s *Storage) Gets(path string, id uint) ([]*Link, error) {
 			if err := s.Delete(link.Hash); err != nil {
 				return nil, err
 			}
-			links = append(links[:i], links[i+1:]...)
+			if len(links) > i+1 {
+				links = append(links[:i], links[i+1:]...)
+			}
```

Comment:
Why this change is introduced?

Reply:
This check doesn't save from data race

Reply:
Sorry, what data race do you mean? It stops accessing an array in an unsafe manner and risking an out of bounds